kraken-build/src/kraken/std/python/buildsystem/uv.py

```diff
@@ -62,136 +63,143 @@ def unsafe_url(self) -> str:
+                env["UV_DEFAULT_INDEX"] = index.unsafe_url
+            else:
+                # https://docs.astral.sh/uv/configuration/environment/#uv_index
+                uv_indexes.append((f"{index.name}=" if index.name != "" else "") + index.unsafe_url)
+
+        if len(uv_indexes) != 0:
+            env["UV_INDEX"] = " ".join(uv_indexes)
         return env
 
 
 class UvPyprojectHandler(PyprojectHandler):
     """Implements the PyprojectHandler interface for UV projects."""
 
-    # TODO: Support `uv.toml` configuration file?
-
-    # PyprojectHandler
+    # TODO: Support global `uv.toml` configuration file?
 
     def get_package_indexes(self) -> list[PackageIndex]:
-        """Maps the UV [`index-url`][1] and [`extra-index-url`][2] options to Kraken's concept of package indices.
-        Note that UV does not support the concept of "aliases" for package indices, so instead the package index alias
-        is ignored and generated automatically based on the hostname and URL hash.
+        """Maps the UV [`index`][1] table, [`index-url`][2] and [`extra-index-url`][3] options to Kraken's concept of
+        package indices. Note that UV does not support the concept of "aliases" for package indices, so instead
+        the package index alias is ignored and generated automatically based on the hostname and URL hash.
 
-        [1]: https://docs.astral.sh/uv/reference/settings/#index-url
-        [2]: https://docs.astral.sh/uv/reference/settings/#extra-index-url
+        [1]: https://docs.astral.sh/uv/reference/settings/#index
+        [2]: https://docs.astral.sh/uv/reference/settings/#index-url
+        [3]: https://docs.astral.sh/uv/reference/settings/#extra-index-url
         """
 
-        def gen_alias(url: str) -> str:
-            hostname = urlparse(url).hostname
-            assert hostname is not None, "expected hostname in package index URL"
-            return f"hostname-{md5(url.encode()).hexdigest()[:5]}"
-
         indexes: list[PackageIndex] = []
-        config: dict[str, Any] = self.raw.get("tool", {}).get("uv", {})
+        for index in self.raw.get("tool", {}).get("uv", {}).get("index", []):
+            indexes.append(
+                PackageIndex(
+                    alias=index.get("name", ""),
+                    index_url=index["url"],
+                    priority=PackageIndex.Priority.default
+                    if index.get("default", False)
+                    else PackageIndex.Priority.explicit
+                    if index.get("explicit", False)
+                    else PackageIndex.Priority.supplemental,
+                    verify_ssl=True,
+                )
+            )
 
-        if index_url := config.get("index-url"):
+        if index_url := self.raw.get("tool", {}).get("uv", {}).get("index-url"):
             indexes.append(
                 PackageIndex(
-                    alias=gen_alias(index_url),
+                    alias="",  # unnamed index
```

Comment:
Might want an optional type for alias instead of that

Reply:
see comment above: https://github.com/kraken-build/kraken/pull/297#discussion_r1925342139